src/Microsoft.ML.FastTree/Representation/TreeRegressor.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Trainers.FastTree.Internal;
+
+namespace Microsoft.ML.FastTree.Representation
+{
+    /// <summary>
+    /// A container class for exposing <see cref="RegressionTree"/>'s attributes to users.
+    /// This class should not be mutable, so it contains a lot of read-only members.
+    /// </summary>
+    public class TreeRegressor
+    {
+        /// <summary>
+        /// <see cref="TreeRegressor"/> is an immutable wrapper over <see cref="_tree"/> for exposing some tree's attribute to users.
+        /// </summary>
+        private readonly RegressionTree _tree;
+
+        /// <summary>
+        /// Sample labels from training data. <see cref="_leafSamples"/>[i] stores the labels falling into the i-th leaf.
+        /// </summary>
+        private readonly List<double[]> _leafSamples;
```

Comment:
>List<double[] [](start = 25, length = 13)

Items in List<double[]> is actually mutable a structure of mutable structures.
Individual list elements and array elements can be modified. Is there a read-only structure we can use here instead?

Reply:
If we change it to `ImmutableArray`, we will have some fields with type `ReadOnlySpan` and two functions returning `ImmutableArray`. This field is private. so maybe we can keep it as it for now?

---
In reply to: [251948660](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251948660) [](ancestors = 251948660)

Reply:
Perhaps turn it into double[][]?

---
In reply to: [251953838](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251953838) [](ancestors = 251953838,251948660)